### PR TITLE
[bitnami/redis] Use definition instead of hardcoded key

### PR DESCRIPTION
**Pull request by uncycler**
### Description of the change

This PR replaces the hardcoded default secret key used to compute the checksum with the defined function.

### Benefits

Allow to use custom secret key name.

### Possible drawbacks

None

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by carrodher** [comment]
Thanks for creating this PR. Please, note it is needed to sign the commits. Your signature certifies that you wrote the patch or otherwise have the right to contribute the material. You can find more info about this topic, in [the “Sign your work” section from the contributing guidelines](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work).

File: bitnami/redis/templates/_helpers.tpl [hunk 1]
```diff
@@ -234,7 +234,7 @@ Return Redis&reg; password
 {{- else if not (empty .Values.auth.password) -}}
     {{- .Values.auth.password -}}
 {{- else -}}
-    {{- include "getValueFromSecret" (dict "Namespace" .Release.Namespace "Name" (include "common.names.fullname" .) "Length" 10 "Key" "redis-password")  -}}
+    {{- include "getValueFromSecret" (dict "Namespace" .Release.Namespace "Name" (include "common.names.fullname" .) "Length" 10 "Key" (template "redis.secretPasswordKey" .))  -}}
```
Comment (jouve) [inline bitnami/redis/templates/_helpers.tpl:237]: I think this needs `include` instead of `template`

Comment (uncycler) [reply bitnami/redis/templates/_helpers.tpl]: It makes more sense yes, thx

**Review by fmulero — COMMENTED** [review]
Hi @uncycler, Thanks for your contribution.

I've been reviewing the change and it makes sense, but I don't fully understand how did you decided to change this piece of code, because the `redis.password` helper seems to be used only when the redis secret is created/update, so using there the `auth.existingSecretPasswordKey`, at first sight, doesn't make sense to me.

**Comment by uncycler** [comment]
Upgrade from 17.7.5 to 17.8.0 fails with the following error:

```
redis/templates/sentinel/statefulset.yaml:50:28: executing "redis/templates/sentinel/statefulset.yaml" at <include (print $.Template.BasePath "/secret.yaml") .>: error calling include: template: redis/templates/secret.yaml:9:17: executing "redis/templates/secret.yaml" at <include "redis.password" .>: error calling include: template: redis/templates/_helpers.tpl:237:8: executing "redis.password" at <include "getValueFromSecret" (dict "Namespace" .Release.Namespace "Name" (include "common.names.fullname" .) "Length" 10 "Key" "redis-password")>: error calling include: template: redis/templates/_helpers.tpl:222:30: executing "getValueFromSecret" at <b64dec>: invalid value; expected string
```

The secret.yaml file is loaded to calculate the checksum in  in which, following 17.8.0 release, include the redis-password definition out of the condition.

**Comment by fmulero** [comment]
Hi @uncycler, sorry to bother you again, but I am not able to reproduce the use case you are trying to fix.

I followed these steps and everything looks good in my side.
```console
$ kubectl create secret generic redis-test  --from-literal=mypass=verystrongpassword
secret/redis-test created
$ helm install redis bitnami/redis --version 17.7.5 --set auth.enabled=true --set auth.existingSecret=redis-test --set auth.existingSecretPasswordKey=mypass
NAME: redis
LAST DEPLOYED: Wed Mar  8 10:05:18 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
CHART NAME: redis
CHART VERSION: 17.7.5
APP VERSION: 7.0.8

** Please be patient while the chart is being deployed **
...
$ helm upgrade redis bitnami/redis --version 17.8.0 --set auth.enabled=true --set auth.existingSecret=redis-test --set auth.existingSecretPasswordKey=mypass
Release "redis" has been upgraded. Happy Helming!
NAME: redis
LAST DEPLOYED: Wed Mar  8 10:06:57 2023
NAMESPACE: default
STATUS: deployed
REVISION: 2
TEST SUITE: None
NOTES:
CHART NAME: redis
CHART VERSION: 17.8.0
APP VERSION: 7.0.8

** Please be patient while the chart is being deployed **

Redis&reg; can be accessed on the following DNS names from within your cluster:

    redis-master.default.svc.cluster.local for read/write operations (port 6379)
    redis-replicas.default.svc.cluster.local for read-only operations (port 6379)



To get your password run:

    export REDIS_PASSWORD=$(kubectl get secret --namespace default redis-test -o jsonpath="{.data.redis-password}" | base64 -d)

To connect to your Redis&reg; server:

1. Run a Redis&reg; pod that you can use as a client:

   kubectl run --namespace default redis-client --restart='Never'  --env REDIS_PASSWORD=$REDIS_PASSWORD  --image docker.io/bitnami/redis:7.0.8-debian-11-r13 --command -- sleep infinity

   Use the following command to attach to the pod:

   kubectl exec --tty -i redis-client \
   --namespace default -- bash

2. Connect using the Redis&reg; CLI:
   REDISCLI_AUTH="$REDIS_PASSWORD" redis-cli -h redis-master
   REDISCLI_AUTH="$REDIS_PASSWORD" redis-cli -h redis-replicas

To connect to your database from outside the cluster execute the following commands:

    kubectl port-forward --namespace default svc/redis-master 6379:6379 &
    REDISCLI_AUTH="$REDIS_PASSWORD" redis-cli -h 127.0.0.1 -p 6379
```
What did I do wrong? Could you share the steps to reproduce the issue?

**Comment by uncycler** [comment]
Try to name the release the same as the secret key:

```
helm install redis-test bitnami/redis --version 17.7.5 --set auth.enabled=true --set auth.existingSecret=redis-test --set auth.existingSecretPasswordKey=mypass
```

It will trigger the error.

The helper is more broken than I though...
```
{{- include "getValueFromSecret" (dict "Namespace" .Release.Namespace "Name" (include "common.names.fullname" .) "Length" 10 "Key" "redis-password")  -}}
```

It should do `.Values.auth.existingSecret | default (include "common.names.fullname" .) ` instead.

File: bitnami/redis/templates/_helpers.tpl [hunk 2]
```diff
@@ -234,7 +234,7 @@ Return Redis&reg; password
 {{- else if not (empty .Values.auth.password) -}}
     {{- .Values.auth.password -}}
 {{- else -}}
-    {{- include "getValueFromSecret" (dict "Namespace" .Release.Namespace "Name" (include "common.names.fullname" .) "Length" 10 "Key" "redis-password")  -}}
+    {{- include "getValueFromSecret" (dict "Namespace" .Release.Namespace "Name" (include "common.names.fullname" .) "Length" 10 "Key" (include "redis.secretPasswordKey" .))  -}}
```
Comment (fmulero) [inline bitnami/redis/templates/_helpers.tpl:237]: According to the root cause of the issue, the proper fix for this function should be something like this:

```yaml
{{/*
Return Redis&reg; password
*/}}
{{- define "redis.password" -}}
{{- if or .Values.auth.enabled .Values.global.redis.password }}
    {{- if not (empty .Values.global.redis.password) }}
        {{- .Values.global.redis.password -}}
    {{- else if not (empty .Values.auth.password) -}}
        {{- .Values.auth.password -}}
    {{- else -}}
        {{- include "getValueFromSecret" (dict "Namespace" .Release.Namespace "Name" (include "redis.secretName" .) "Length" 10 "Key" (include "redis.secretPasswordKey" .))  -}}
    {{- end -}}
{{- end -}}
{{- end }}
```

**Pull request closed without merging**